### PR TITLE
Snippets for SqlServer 3.1

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@SimonCropp is there a better option than to copy all the contents of 3.0 snippets to create 3.1 snippets project?

**Comment by SimonCropp** [comment]
@SzymonPobiega if u dont want the drop down u can just put `3.1` at the end of the snippets that are new.

**Comment by SzymonPobiega** [comment]
How would the partials work in this case?

**Comment by SimonCropp** [comment]
partials have to target a dir. so u would need the snippet dir split for that.

or u can make the partial target 6.0 and add a note about the applicable version. often if the partial speaks to a nearby snippet with the specific version the the the "note about the applicable version" is redundant.

remember u can merge to master and iterate on the outcome. u dont have to get it right the first time

File: Snippets/SqlServer/SqlServer.sln [hunk 1]
```diff
@@ -1,37 +1,55 @@
...
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SqlServer_2.1", "SqlServer_2.1\SqlServer_2.1.csproj", "{DB7B97B4-3075-4541-A821-1C6FF9AB8671}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SqlServer_2.1", "SqlServer_2.1\SqlServer_2.1.csproj", "{DB7B97B4-3075-4541-A821-1C6FF9AB8671}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SqlServer_3.1", "SqlServer_3.1\SqlServer_3.1.csproj", "{E8130F3D-3346-4846-9122-2406EB4E31E4}"
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
```
Comment (MarcinHoppe) [inline Snippets/SqlServer/SqlServer.sln]: I don't think we have `Release` configuration for other snippet solutions?

File: Snippets/SqlServer/SqlServer_3.1/MultiCatalog.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    void CatalogForQueue(EndpointConfiguration endpointConfiguration, IMessageSession messageSession)
+    {
+        #region sqlserver-multicatalog-config-for-queue
+
+        var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+
+        transport.UseCatalogForQueue(queueName: "myqueue", catalog: "MyCatalog");
+        transport.UseCatalogForQueue(queueName: "error", catalog: "ServiceControl");
+
+        messageSession.Send("myqueue", new MyMessage());
```
Comment (MarcinHoppe) [inline Snippets/SqlServer/SqlServer_3.1/MultiCatalog.cs]: It seems odd that this is right after configuring the transport. Maybe put ellipsis between those two sections to indicate they are not connected to each other or split sending into separate snippet altogether.

File: Snippets/SqlServer/SqlServer_3.1/RuntimeScripts/CreateQueueText.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+                    IF NOT  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'[{0}].[{1}]') AND type in (N'U'))
+                    BEGIN
+                        CREATE TABLE [{0}].[{1}](
+                            [Id] [uniqueidentifier] NOT NULL,
+                            [CorrelationId] [varchar](255) NULL,
+                            [ReplyToAddress] [varchar](255) NULL,
+                            [Recoverable] [bit] NOT NULL,
+                            [Expires] [datetime] NULL,
+                            [Headers] [varchar](max) NOT NULL,
```
Comment (MarcinHoppe) [inline Snippets/SqlServer/SqlServer_3.1/RuntimeScripts/CreateQueueText.sql]: Use `nvarchar` for headers.

File: Snippets/SqlServer/SqlServer.sln (hunk 1 above)
Comment (SimonCropp) [inline Snippets/SqlServer/SqlServer.sln]: yeah a new update of VS 2017 seems to re add it automatically now. so dont sweat it

**Pull request merged**